crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -501,6 +558,29 @@ impl RoomEventCache {
         None
     }
 
+    /// Try to find an event by id in this room, along with all relations.
+    pub async fn event_with_relations(&self, event_id: &EventId) -> Vec<SyncTimelineEvent> {
```

Comment:
Can you return an `(SyncTimelineEvent, Vec<SyncTimelineEvent>)` instead, so it's clear where the target event is? (Right now there's nothing either in the signature or the doc-comment that makes it obvious it's the first one.)